Add database env config documentation (#1504)
strapi/documentation: docs/developer-docs/latest/setup-deployment-guides/configurations/required/databases.md

| Parameter  | Description                                                                                                                   | Type                  |
|------------|-------------------------------------------------------------------------------------------------------------------------------|-----------------------|

> | `connectionString`| Database connection string. When set, it overrides the other `connection.connection` properties. To disable use an empty string: `''`. Available in v4.6.2+                    | `String`                  |

Reviewer: I think we could use additional formatting to make the "Available in […]" stand out.
For consistency, I would also suggest we use the same formatting everywhere in the table (see line 55, "Used only […]")
Adding a line break and using italics should be enough. What do you think?

Author: OK with me. I used bold since that is what was on line 55.

});
```

## Environment variables in database configurations

> Strapi version v4.6.2 and higher includes the database configuration options in the `./config/database.js` or `./config/datbase.ts` file. When a new project is created the environment variable `DATABASE_CLIENT` with the value `mysql`, `postgres`, or `sqlite` is automatically added to the `.env` file depending on which database you choose during project creation. Additionally, all of the environment variables necessary to connect to your local development database are also added to the `.env` file.  The following is an example of the generated config file:

Reviewer: ```suggestion
Strapi version v4.6.2 and higher includes the database configuration options in the `./config/database.js` or `./config/datbase.ts` file. When a new project is created the environment variable `DATABASE_CLIENT` with the value `mysql`, `postgres`, or `sqlite` is automatically added to the `.env` file depending on which database you choose during project creation. Additionally, all of the environment variables necessary to connect to your local development database are also added to the `.env` file.  The following is an example of the generated configuration file:
```
FYI, Grammarly also suggests the following improvements 👀 :
![Screenshot 2023-02-15 at 14 05 41](https://user-images.githubusercontent.com/4233866/219035508-5bfb2a6b-6215-4628-a30d-c14f689aef7d.png)

        database: env('DATABASE_NAME', 'strapi'),
        user: env('DATABASE_USERNAME', 'strapi'),
        password: env('DATABASE_PASSWORD', 'strapi'),
        ssl: env.bool('DATABASE_SSL', false) && {
          key: env('DATABASE_SSL_KEY', undefined),
          cert: env('DATABASE_SSL_CERT', undefined),
          ca: env('DATABASE_SSL_CA', undefined),
          capath: env('DATABASE_SSL_CAPATH', undefined),
          cipher: env('DATABASE_SSL_CIPHER', undefined),
          rejectUnauthorized: env.bool(
            'DATABASE_SSL_REJECT_UNAUTHORIZED',
            true
          ),
        },
        schema: env('DATABASE_SCHEMA', 'public'),
      },
      pool: { min: env.int('DATABASE_POOL_MIN', 2), max: env.int('DATABASE_POOL_MAX', 10) },
    },
    sqlite: {
      connection: {
        filename: path.join(
          __dirname,
          '..',
          env('DATABASE_FILENAME', 'data.db')
        ),
      },
      useNullAsDefault: true,
    },
  };

  return {
    connection: {
      client,
      ...connections[client],
      acquireConnectionTimeout: env.int('DATABASE_CONNECTION_TIMEOUT', 60000),
    },
  };
};
```

</code-block>
</code-group>

> The following are examples of the corresponding `.env` file `# Database` section for each of the possible databases:

Reviewer: ```suggestion
The following are examples of the corresponding `.env` file database-related keys for each of the possible databases:
```

I suggest we remove the formatting and slightly update the wording since `# Database` is just a comment in the `.env` file. Technically it's not even really a section. These lines could be placed anywhere in .env.

<code-group>
<code-block title="MySQL">

```bash

# Database
DATABASE_CLIENT=mysql
DATABASE_HOST=127.0.0.1
DATABASE_PORT=3306
DATABASE_NAME=strapi
DATABASE_USERNAME=strapi
DATABASE_PASSWORD=strap1
DATABASE_SSL=false
```

</code-block>

<code-block title="PostgreSQL">

```bash

# Database
DATABASE_CLIENT=postgres
DATABASE_HOST=127.0.0.1
DATABASE_PORT=5432
DATABASE_NAME=strapi
DATABASE_USERNAME=strapi
DATABASE_PASSWORD=strapi
DATABASE_SSL=false
```

</code-block>

<code-block title="SQLite">

```bash

# Database
DATABASE_CLIENT=sqlite
DATABASE_FILENAME=.tmp/data.db
```

</code-block>
</code-group>

> :::strapi

Reviewer: Let's add a title for the callout :)

Author: n/a since the callout is now gone

If you started your project with a version prior to v4.6.2 you can convert your `database.js|database.ts` configuration file following this procedure:

1. Update your application to v4.6.2 or a later version. See the [Updates and Migrations](/developer-docs/latest/update-migration-guides/migration-guides/migration-guides.md) section for any breaking changes that require specific fixes.
2. Replace the contents of your `./config/database.js` or `./config/database.ts` file with the preceding JavaScript or TypeScript code.
3. Add the environment variables from the preceding code example to your `.env` file.
4. (_optional_) Add additional environment variables such as `DATABASE_URL` and the properties of the `ssl` object.
5. Save the changes and restart your application.

> ::::caution

Reviewer: I think we should avoid embedding a callout in another callout. I'd say:
- Either the caution callout is really important: In that case, I'd suggest turning the whole "strapi" callout into a section. This section's title (h3) would be the title you would add to the strapi callout (see previous comment).
- Or we extract the sentence from the caution callout and highlight it (bold formatting?).

Since this page is getting longer and longer, I would go for the first solution. Can you think of another solution? 🤔

Author: Yes the first solution sounds good.

Do not overwrite the environment variables: `HOST`, `PORT`, `APP_KEYS`, `API_TOKEN_SALT`, and `ADMIN_JWT_SECRET`.
::::
:::

### Database connections using `connectionString`

Many managed database solutions use the property `connectionString` to connect a database to an application. Strapi v4.6.2 and later versions include the `connectionString` property. The `connectionString` is a concatenation of all the database properties in the `connection.connection` object. The `connectionString`:

- overrides the other `connection.connection` properties such as `host` and `port`,
- can be disabled by setting the property to an empty string: `''`.

### Database management by environment

> Development of a Strapi application commonly includes customization in the local development environment with a local development database, such as `SQLite`. When the application is ready for another environment such as production or staging the application is deployed with a different database instance, usually `MySQL`, `MariaDB`, or `PostgreSQL`. Database environment variables allow you to switch the attached database. To switch the database connection set a minimum of the `DATABASE_CLIENT` and `DATABASE_URL` for `MySQL`, `MariaDB`, and `PostgreSQL`. Set a minimum of `DATABASE_CLIENT` and `DATABASE_FILENAME` for `SQLite`.

Reviewer: ```suggestion
Development of a Strapi application commonly includes customization in the local development environment with a local development database, such as `SQLite`. When the application is ready for another environment such as production or staging the application is deployed with a different database instance, usually `MySQL`, `MariaDB`, or `PostgreSQL`. Database environment variables allow you to switch the attached database. To switch the database connection:
* set a minimum of the `DATABASE_CLIENT` and `DATABASE_URL` for `MySQL`, `MariaDB`, and `PostgreSQL`,
* or set a minimum of `DATABASE_CLIENT` and `DATABASE_FILENAME` for `SQLite`.
```

unless I misunderstood this part 🤔
